subsys/bluetooth/host/hci_core.c

```diff
@@ -2927,13 +2928,16 @@ static const struct event_handler normal_events[] = {
  * Helper for the HCI driver to get HCI event flags that describes rules that.
  * must be followed.
  *
- * @param evt HCI event code.
+ * @param evt The first 3 bytes of the HCI event. The first byte
+ * is the event code. The second is the length. The third is the
+ * first byte of the event payload. Assertion: There are no
+ * zero-length event payloads.
  *
  * @return HCI event flags for the specified event.
  */
-static inline uint8_t bt_hci_evt_get_flags(uint8_t evt)
+static inline uint8_t bt_hci_evt_get_flags(uint8_t evt[static 3])
```

Comment:
I've never seen the `static` keyword used in an array size designator like that. What kind of sorcery is this?

Reply:
And more importantly, which C standard is this from, since there's a risk that it may introduce some compatibility issues (see https://docs.zephyrproject.org/latest/develop/languages/c/index.html#language-standards)

Reply:
@alwa-nordic is the C language lawyer lol.

another nugget was we had `while (num_reports --> 0)`. Which is also valid 🙃

Reply:
Looks like it's from C11, so we should be fine there: https://stackoverflow.com/questions/3430315/what-is-the-purpose-of-static-keyword-in-array-parameter-of-function-like-char

Reply:
https://hamberg.no/erlend/posts/2013-02-18-static-array-indices.html :)

Reply:
> @alwa-nordic is the C language lawyer lol.
> 
> another nugget was we had `while (num_reports --> 0)`. Which is also valid 🙃

perfectly valid code: `value += (value = value == value --> value <-- value), value++;` I see myself out now